include/ultra64/viint.h

```diff
@@ -17,12 +17,12 @@
 // For use in initializing OSViMode structures
 
 #define BURST(hsync_width, color_width, vsync_width, color_start) \
-    (hsync_width | (color_width << 8) | (vsync_width << 16) | (color_start << 20))
+    ((u8)(hsync_width) | ((u8)(color_width) << 8) | ((u8)(vsync_width) << 16) | ((u16)(color_start) << 20))
 #define WIDTH(v) v
 #define VSYNC(v) v
 #define HSYNC(duration, leap) (duration | (leap << 16))
-#define LEAP(upper, lower) ((upper << 16) | lower)
-#define START(start, end) ((start << 16) | end)
+#define LEAP(upper, lower) (((u16)(upper) << 16) | (u16)(lower))
+#define START(start, end) (((u16)(start) << 16) | (u16)(end))
```

Comment:
It *might* be better style to just cast everything to u32 (including the ones that don't require it, like leap), since shifts are involved, and casting to u32 should never generate additional ands if there are variables involved.

Reply:
I think there are never variables involves in those macros so I don't think it is something we should worry about